realm/realm-library/src/main/java/io/realm/internal/NativeObjectReference.java

```diff
@@ -23,43 +23,70 @@
  * This class is used for holding the reference to the native pointers present in NativeObjects.
  * This is required as phantom references cannot access the original objects for this value.
  */
-public final class NativeObjectReference extends PhantomReference<NativeObject> {
+final class NativeObjectReference extends PhantomReference<NativeObject> {
 
-    // Using int here instead of enum to make it faster since the cleanup needs to be called
-    // in a loop to dealloc every native reference.
-    public static final int TYPE_LINK_VIEW = 0;
-    public static final int TYPE_ROW = 1;
+    // Linked list to keep the reference of the PhantomReference
```

Comment:
I think i tried `java.util.LinkedList`, but that was much slower than this custom implementation, mainly because of it adds overhead by some extra features we don't need here.